corehq/apps/app_manager/views/modules.py

```diff
@@ -488,6 +489,11 @@ def rename_action_case_type(mod):
             except ModuleNotFoundException:
                 messages.error(_("Unknown Module"))
 
+    if should_edit('incl_form_ids') and isinstance(module, ShadowModule):
+        incl = request.POST.getlist('incl_form_ids') or []
+        excl = {f.unique_id for f in module.source_module.get_forms()} - set(incl)
+        module.excluded_form_ids = list(excl)
```

Comment:
Yeah, in that scenario it would be better to work out the excluded forms on the client side, and pass `excl_form_ids` to the server ... 

However, you could argue (but only if you were feeling argumentative :grin:) that the user should expect the new form not to be included, because they didn't select it a moment ago in the list of forms (because it wasn't there) -- so although the behaviour isn't ideal, it is at least obvious.

I agree that it would be optimal to include those new forms in order to be consistent. I'm tempted to move lines 493 and 494 to JS ... but I also love merged PRs. I'll try to move them before you get into the office.